src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -191,7 +191,7 @@ def _get_azure_cli_properties(self):
                               lambda: _get_config().get('core', 'output', fallback='unknown'))
         set_custom_properties(result, 'EnvironmentVariables', _get_env_string)
         set_custom_properties(result, 'Locale',
-                              lambda: '{},{}'.format(locale.getdefaultlocale()[0], locale.getdefaultlocale()[1]))
+                              lambda: '{},{}'.format(locale.getdefaultlocale()[0], locale.getdefaultlocale()[1]))  # pylint:disable=deprecated-method
```

Comment:
`locale.getdefaultlocale()` is replaced by `locale.getlocale()`, but the result is different on Windows:
```
>>> locale.getlocale()
('English_United States', '1252')
>>> locale.getdefaultlocale()
('en_US', 'cp1252')
```
I can use `locale.getencoding()`(available in 3.11) to get `cp1252`, but I don't know how to get `en_US`.
@evelyn-ys for awareness.

Reply:
I think simply replacing with `locale.getlocale()` is acceptable since this property is not used in any report and the value change is not a big deal

Reply:
Linux doesn't have this inconsistency issue:

```py
$ python -c "import locale; print(locale.getlocale()); print(locale.getdefaultlocale())"
('en_US', 'UTF-8')
('en_US', 'UTF-8')
```

This "`getlocale` returning `English_United States` on Windows" issue is reported by https://github.com/python/cpython/issues/82986.

Reply:
Linux doesn't have this inconsistency issue:

```py
$ python -c "import locale; print(locale.getlocale()); print(locale.getdefaultlocale())"
('en_US', 'UTF-8')
('en_US', 'UTF-8')
```

This "`getlocale` returning `English_United States` on Windows" issue is reported by https://github.com/python/cpython/issues/82986.

I think this is because [`locale.setlocale`](https://github.com/python/cpython/blob/main/Modules/_localemodule.c#L101) internally calls [`setlocale`](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/setlocale-wsetlocale) C API, while [`locale.getdefaultlocale`](https://github.com/python/cpython/blob/main/Modules/_localemodule.c#L444) calls [`GetLocaleInfoA`](https://learn.microsoft.com/en-us/windows/win32/api/winnls/nf-winnls-getlocaleinfoa) C API. [`setlocale`](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/setlocale-wsetlocale) C API also mentions using `English_United States` on Windows.